### ISSUE TITLE
MA0028 should not be reported when fix involves boxing

### ISSUE THREAD

**Issue by drieseng** [meziantou/Meziantou.Analyzer#364]
**Context**
Rule identifier: MA0028
Target Framework: _.NET 4.8_
Language version: _10.0_

**Describe the bug**
Right now, **MA0028** is reported when `StringBuilder.Append(string)` is invoked with as argument the return value of a  `ToString(string, IFormatProvider)` invocation.

This is ok when the instance on which `ToString(string, IFormatProvider)` is invoked is a reference type. When it's a value type, then changing the call to `StringBuilder.AppendFormat(IFormatProvider provider, string format, object arg0)` would involve boxing.

**To Reproduce**
Code or steps to reproduce the behavior:

````c#
var now = DateTime.Now;
= sb.Append(now.ToString("s", CultureInfo.InvariantCulture));
````

**MA0028** is reported, and the following code is produced if you apply the code fix:
```c#
var now = DateTime.Now;
= sb.AppendFormat(CultureInfo.InvariantCulture, "{0:s}", now);
```

The proposed code now involves boxing.

**Expected:**
**MA0028** should not be reported if the instance on which `ToString(string, IFormatProvider)` is invoked is a value type.

**Personal note:**
From a performance point-of-view, I don't see why you would ever prefer `StringBuilder.AppendFormat(CultureInfo.CurrentCulture, "{0:n1}", 10)` over `StringBuilder.Append(something.ToString("n1", CultureInfo.CurrentCulture))`.

I would only propose to use:
* `StringBuilder.AppendFormat(IFormatProvider provider, string format, object arg0, object arg1, ...)` over `StringBuilder.Append(String.Format(IFormatProvider provider, string format, object arg0, object arg1, ...))` (and similar overloads of `String.Format(IFormatProvider provider, string format, ...)`)
* `StringBuilder.AppendFormat(string format, object arg0, object arg1, ...)` over `StringBuilder.Append(string.Format(string format, object arg0, object arg1, ...))` (and similar overloads of `String.Format(string format, ...)`)

**Comment by meziantou** [comment]
I agree! I'll update the code to report `string.Format` case: https://github.com/meziantou/Meziantou.Analyzer/pull/366
